core/src/main/java/org/opensearch/sql/expression/ip/IPFunction.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.expression.ip;
+
+import static org.opensearch.sql.data.type.ExprCoreType.BOOLEAN;
+import static org.opensearch.sql.data.type.ExprCoreType.STRING;
+import static org.opensearch.sql.expression.function.FunctionDSL.define;
+import static org.opensearch.sql.expression.function.FunctionDSL.impl;
+import static org.opensearch.sql.expression.function.FunctionDSL.nullMissingHandling;
+
+import inet.ipaddr.AddressStringException;
+import inet.ipaddr.IPAddressString;
+import inet.ipaddr.IPAddressStringParameters;
+import lombok.experimental.UtilityClass;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.data.model.ExprValueUtils;
+import org.opensearch.sql.exception.SemanticCheckException;
+import org.opensearch.sql.expression.function.BuiltinFunctionName;
+import org.opensearch.sql.expression.function.BuiltinFunctionRepository;
+import org.opensearch.sql.expression.function.DefaultFunctionResolver;
+
+/** Utility class that defines and registers IP functions. */
+@UtilityClass
+public class IPFunction {
+
+  public void register(BuiltinFunctionRepository repository) {
+    repository.register(cidrmatch());
+  }
+
+  private DefaultFunctionResolver cidrmatch() {
+
+    // TODO #3145: Add support for IP address data type.
+    return define(
+        BuiltinFunctionName.CIDRMATCH.getName(),
+        impl(nullMissingHandling(IPFunction::exprCidrMatch), BOOLEAN, STRING, STRING));
+  }
+
+  /**
+   * Returns whether the given IP address is within the specified CIDR IP address range. Supports
```

Comment:
Would be nice to add, but I see https://github.com/opensearch-project/opensearch-spark/blob/main/docs/ppl-lang/functions/ppl-ip.md doesn't have this
```suggestion
   * Returns whether the given IP address is within the specified CIDR (Classless Inter-Domain Routing) IP address range. Supports
```

Reply:
@acarbonetto good idea - lets try to sync the PPL docs - at least until we unify them into a common location

Reply:
Gonna hold off on adding this for now if that's OK, just to keep documentation consistent (though agree that this is more clear).

Reply:
These changes can be made in a separate PR.